iCloud Photo Library Downloading

jlalvarez18 opened the pull request:
- added getMediaFilesBetween() to fetch between dates
- updated example to work with latest XCode and get files for the month
- introduced MediaAsset

Sauraus commented on example/.flutter-plugins-dependencies:
```diff
@@ -0,0 +1 @@
+{"info":"This is a generated file; do not edit or check into version control.","plugins":{"ios":[{"name":"permission_handler","path":"/Users/juan/dev/flutter/.pub-cache/hosted/pub.dartlang.org/permission_handler-3.1.0/","dependencies":[]},{"name":"media_picker_builder","path":"/Users/juan/private_repos/media_picker_builder/","dependencies":[]}],"android":[{"name":"permission_handler","path":"/Use ...
```
Please commit a proper .gitignore generated at https://gitignore.io

Sauraus commented on example/ios/Flutter/.last_build_id:
```diff
@@ -0,0 +1 @@
+19ac9e0be5b5e961faee53688bb5943d
```
Not sure this file belongs in git.

Sauraus reviewed (changes requested):
@jlalvarez18 lets do a call with @qbait, @adamstyrc and myself on Monday to walk,through these changes, there is a lot here.

jlalvarez18 committed:
updated MediaFetcher

- added sort and media types handling to getAssets()
- now returning proper orientation when fetching MediaFiles
- updated gitignore
- using flutter 1.22
- updated example project

jlalvarez18 committed:
now distinguishing between assets and files

- added MediaAsset class

jlalvarez18 committed:
more enhancements to MediaFetcher (live photos)

- re-created ios project using flutter 1.22.2

jlalvarez18 committed:
fixed issues with date range

- added boolean to include/ignore live photos
- added types arugment to getMediaAssets

jlalvarez18 committed:
added isLivePhoto to MediaAsset and MediaFile

jlalvarez18 committed:
added progress handler for retrieveMediaFile

Sauraus commented on .gitignore:
```diff
@@ -1,7 +1,62 @@
-.DS_Store
-.dart_tool/
+# Created by https://www.toptal.com/developers/gitignore/api/flutter
+# Edit at https://www.toptal.com/developers/gitignore?templates=flutter
```
We probably want this to be 

https://www.toptal.com/developers/gitignore?templates=dart,swift,kotlin,flutter,intellij,androidstudio,visualstudiocode,osx,windows

jlalvarez18 replied:
Nice!

Sauraus reviewed (changes requested):
What about the albumname that @qbait removed entirely from the code base, did you come across that and why it was giving us grief?

jlalvarez18 commented:
@Sauraus album name? My new re-write doesn't care about albums. You fetch based on type and date range. The old codebase used albums, then iterated through the albums, then fetched files from the cloud whether you liked it or not, then returned you the results.

jlalvarez18 committed:
updated .gitignore

Sauraus commented:
> @Sauraus album name? My new re-write doesn't care about albums. You fetch based on type and date range. The old codebase used albums, then iterated through the albums, then fetched files from the cloud whether you liked it or not, then returned you the results.

So what if I want to import from an album that's no longer possible?
